packages/flutter_tools/bin/macos_assemble.sh

```diff
@@ -62,26 +62,22 @@ ephemeral_dir="${SOURCE_ROOT}/Flutter/ephemeral"
 build_inputs_path="${ephemeral_dir}/FlutterInputs.xcfilelist"
 build_outputs_path="${ephemeral_dir}/FlutterOutputs.xcfilelist"
 
-icon_tree_shaker_flag="false"
-if [[ -n "$TREE_SHAKE_ICONS" ]]; then
-  icon_tree_shaker_flag="true"
-fi
-
-dart_obfuscation_flag="false"
-if [[ -n "$DART_OBFUSCATION" ]]; then
-  dart_obfuscation_flag="true"
+local performance_measurement_option=""
+if [[ -n "$PERFORMANCE_MEASUREMENT_FILE" ]]; then
+  performance_measurement_option="--performance-measurement-file=${PERFORMANCE_MEASUREMENT_FILE}"
 fi
 
 RunCommand "${FLUTTER_ROOT}/bin/flutter"                                    \
     ${verbose_flag}                                                         \
     ${flutter_engine_flag}                                                  \
     ${local_engine_flag}                                                    \
     assemble                                                                \
+    ${performance_measurement_option}                                       \
     -dTargetPlatform=darwin-x64                                             \
     -dTargetFile="${target_path}"                                           \
     -dBuildMode="${build_mode}"                                             \
-    -dTreeShakeIcons="${icon_tree_shaker_flag}"                             \
-    -dDartObfuscation="${dart_obfuscation_flag}"                            \
+    -dTreeShakeIcons="${TREE_SHAKE_ICONS}"                                  \
```

Comment:
There is no need to conditionally set these flags, assemble just checks if `== true`, so an empty string if they are unset is fine.